index.js

```diff
@@ -1,14 +1,23 @@
 'use strict';
+
 require('harmony-reflect');
 
-module.exports = function (obj) {
-	return new Proxy(obj, {
-		get: function (target, name, receiver) {
+module.exports = function (target) {
+	var handler = {
+		get: function (target, name) {
 			if (!(name in target)) {
 				throw new TypeError('Unknown property: ' + name);
 			}
 
-			return Reflect.get(target, name, receiver);
+			return Reflect.get(target, name);
 		}
-	});
+	};
+
+	var instance = new Proxy(target, handler);
+
+	instance.decorateWith = function (fn) {
```

Comment:
What I'm saying is that the usage that we are implementing here, will only lock down "static" properties of the class. I don't know if #1 wants that or instance properties.